src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -934,6 +1116,205 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
+                           chip::EndpointId endpointId = kInvalidEndpointId;
+                           chip::ClusterId clusterId = kInvalidClusterId;
+                           chip::AttributeId attributeId = kInvalidAttributeId;
+
+                           if ([attributePaths[i] endpoint]) {
+                               endpointId = static_cast<chip::EndpointId>([[attributePaths[i] endpoint] unsignedShortValue]);
+                           }
+
+                           if ([attributePaths[i] cluster]) {
+                               clusterId = static_cast<chip::ClusterId>([[attributePaths[i] cluster] unsignedLongValue]);
+                           }
+
+                           if ([attributePaths[i] attribute]) {
+                               attributeId = static_cast<chip::AttributeId>([[attributePaths[i] attribute] unsignedLongValue]);
+                           }
+
+                           attributePathParamsList.push_back(app::AttributePathParams(endpointId, clusterId, attributeId));
+                       }
+                       container.pathParamsSize = attributePathParamsList.size();
+                       container.pathParams = attributePathParamsList.data();
+                   }
+                   if (eventPaths != nil) {
+                       for (uint8_t i = 0; i < [eventPaths count]; i++) {
+                           chip::EndpointId endpointId = kInvalidEndpointId;
+                           chip::ClusterId clusterId = kInvalidClusterId;
+                           chip::EventId eventId = kInvalidEventId;
+
+                           if ([eventPaths[i] endpoint]) {
+                               endpointId = static_cast<chip::EndpointId>([[eventPaths[i] endpoint] unsignedShortValue]);
+                           }
+
+                           if ([eventPaths[i] cluster]) {
+                               clusterId = static_cast<chip::ClusterId>([[eventPaths[i] cluster] unsignedLongValue]);
+                           }
+
+                           if ([eventPaths[i] event]) {
+                               eventId = static_cast<chip::EventId>([[eventPaths[i] event] unsignedLongValue]);
+                           }
+
+                           eventPathParamsList.push_back(app::EventPathParams(endpointId, clusterId, eventId));
+                       }
+                       container.eventPathParamsSize = eventPathParamsList.size();
+                       container.eventPathParams = eventPathParamsList.data();
+                   }
+
+                   app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
+                   CHIP_ERROR err = CHIP_NO_ERROR;
+
+                   chip::app::ReadPrepareParams readParams(session.Value());
+                   [params toReadPrepareParams:readParams];
+                   readParams.mpAttributePathParamsList = container.pathParams;
+                   readParams.mAttributePathParamsListSize = container.pathParamsSize;
```

Comment:
If this is the only use of `pathParamsSize` and `eventPathParamsSize`, why do we need them on the container anyway?  I don't think we need them there...  Again, te comtainer is just there to manage the array lifetime, and for that it does not need to know the size.

Reply:
I understand. I removed.